docs/docs/Contributing/contributing-how-to-contribute.md

```diff
@@ -4,157 +4,110 @@ sidebar_position: 1
+As an open-source project in a rapidly developing field, we are extremely open
+to contributions, whether it be in the form of a new feature, improved infra, or better documentation.
 
-:::info
+To contribute to this project, please follow the [fork and pull request](https://docs.github.com/en/get-started/quickstart/contributing-to-projects) workflow.
 
-This page may contain outdated information. It will be updated as soon as possible.
+## Prerequisites
 
-:::
+- [uv(>=0.4)](https://docs.astral.sh/uv/getting-started/installation/)
+- [Node.js](https://nodejs.org/en/download/package-manager)
 
+## Contribute code
 
+Develop Langflow locally with [uv](https://docs.astral.sh/uv/getting-started/installation/) and [Node.js](https://nodejs.org/en/download/package-manager).
 
+### Clone the Langflow Repository
 
-👋 Hello there! 
+1. Navigate to the [Langflow GitHub repository](https://github.com/langflow-ai/langflow) and click **Fork**.
 
-We welcome contributions from developers of all levels to our open-source project on [GitHub](https://github.com/langflow-ai/langflow). If you'd like to contribute, please check our contributing guidelines and help make Langflow more accessible.
+2. Add the new remote to your local repository on your local machine:
 
+```bash
+git remote add fork https://github.com/<your_git_username>/langflow.git
+```
 
+### Prepare the development environment
 
-As an open-source project in a rapidly developing field, we are extremely open to contributions, whether in the form of a new feature, improved infra, or better documentation.
+1. Create development hooks.
 
+```bash
+make init
+```
 
+This command sets up the development environment by installing backend and frontend dependencies, building the frontend static files, and initializing the project. It runs `make install_backend`, `make install_frontend`, `make build_frontend`, and finally `uv run langflow run` to start the application.
 
-To contribute to this project, please follow a ["fork and pull request"](https://docs.github.com/en/get-started/quickstart/contributing-to-projects) workflow. Please do not try to push directly to this repo unless you are a maintainer.
+2. Run `make lint`, `make format`, and `make unit_tests` before pushing to the repository.
 
+### Debug
 
-## Local Development {#0388cc3c758d434d994022863a6bafa9}
+The repo includes a `.vscode/launch.json` file for debugging the backend in VSCode, which is a lot faster than debugging with Docker compose. To debug Langflow with the `launch.json` file in VSCode:
 
+1. Open Langflow in VSCode.
+2. Press Ctrl+Shift+D (or Cmd+Shift+D on Mac) to open the Run and Debug view.
+3. Choose a configuration from the dropdown at the top (for example, "Debug Backend").
+4. Click the green play button or press F5 to start debugging.
+5. This allows you to quickly debug different parts of your application, like the backend, frontend, or CLI, directly from VSCode.
```

Comment:
This is informational, not an action that has to be taken, so it should not be a numbered step.